orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -229,12 +230,16 @@ def _get_session_config(self):
             providers = (["ROCMExecutionProvider"] if self.is_rocm_pytorch else [
                          "CUDAExecutionProvider"])
             providers.append("CPUExecutionProvider")
+            cuda_provider_option = {"device_id": str(self._device.index)}
+            if not self.is_rocm_pytorch:
+                # Set Conv algo search mode to HEURISTIC, which is same as PyTorch's default setting.
+                cuda_provider_option["cudnn_conv_algo_search"] = "HEURISTIC"
+                cuda_provider_option["cudnn_conv_use_max_workspace"] = "1"
             if self._use_external_gpu_allocator:
-                provider_options = [{"device_id": str(self._device.index),
-                                     "gpu_external_alloc": str(self._torch_alloc),
-                                     "gpu_external_free": str(self._torch_free)}, {}]
-            else:
-                provider_options = [{"device_id": str(self._device.index)}, {}]
+                cuda_provider_option["gpu_external_alloc"] = str(self._torch_alloc)
```

Comment:
nit, this should still be named as provider_options, since gpu_external_* options still apply to rocm EP.